Algorithms/Divide-and-Conquer/MedianFinding.cpp

```diff
@@ -0,0 +1,110 @@
+// This program finds median of a given array
+
+#include<bits/stdc++.h>
+using namespace std;
+#define sz(a)	int(a.size())
+#define pb 	 	push_back
+
+// function for finding median
+int findMedian (int array[], int size, int position)
+{
+	//maximum number of partitions 
+	int partition = (size+4)/5;				
+	std::vector<int> sublist[partition];
+	int i, j;
+	
+	// dividing the array into partitions
+	for (i = 0; i < size; i+=5)
+	{
+		for (j = i; j < i+5 && j < size; ++j)
+		{
+			sublist[i/5].pb(array[j]);				
+		}
+		sort(sublist[i/5].begin() ,sublist[i/5].end());
+	}
+	
+	// median stores the median of each partition
+	int median[partition];						
+	
+	for (i = 0; i < partition; ++i)
+	{
+		int size_of_sublist = sz(sublist[i]);
+		median[i] = sublist[i][size_of_sublist/2];
+	}
+	
+	int pivot;
+	// finding pivot
+	if(partition <= 5)
+	{
+		// if size is small then sorting takes linear time so simply sort and find median
+		sort(median, median + partition);
+		pivot = median[(partition-1)/2];
+	}
+	else
+	{
+		//as size of median array is large find median of medians
+		pivot = findmedian(median, partition, partition/2);
```

Comment:
@h-sinha This call would not work since you have renamed the function. Please run your program and make changes.